packages/cli/create-strapi-starter/utils/build-starter.js

```diff
@@ -119,26 +126,25 @@ module.exports = async function buildStarter(programArgs, program) {
   // Copy the downloaded frontend folder to the project folder
   const frontendPath = join(rootPath, 'frontend');
 
-  const starterDir = (await fse.pathExists(join(tmpDir, 'starter'))) ? 'starter' : 'frontend';
-
   try {
-    await fse.copy(join(tmpDir, starterDir), frontendPath, {
+    await fse.copy(join(starterPath, 'starter'), frontendPath, {
       overwrite: true,
       recursive: true,
     });
   } catch (error) {
     stopProcess(`Failed to create ${chalk.yellow(frontendPath)}: ${error.message}`);
   }
 
-  // Delete temporary directory
-  await fse.remove(tmpDir);
+  // Delete the starter directory if it was downloaded
+  if (!isLocalStarter) {
+    await fse.remove(starterParentPath);
+  }
 
-  const fullUrl = `https://github.com/${fullName}`;
   // Set command options for Strapi app
   const generateStrapiAppOptions = {
     ...program,
-    starter: fullUrl,
-    template: starterJson.template,
+    starter: starterPackageInfo.name,
+    template: `${starterJson.template.name}@${starterJson.template.version}`,
```

Comment:
You could handle the case no template version is provided so it just load the latest template version from npm. Even if not recommended there is no good reason to prevent this